src/shared/regions/defaultRegionProvider.ts

```diff
@@ -4,98 +4,59 @@
  */
 
 import * as vscode from 'vscode'
-import { endpointsFileUrl } from '../constants'
-import { ext } from '../extensionGlobals'
-import { getLogger, Logger } from '../logger'
-import { CompositeResourceFetcher } from '../resourcefetcher/compositeResourceFetcher'
-import { FileResourceFetcher } from '../resourcefetcher/fileResourceFetcher'
-import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
-import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { Endpoints, Region } from './endpoints'
+import { EndpointsProvider } from './endpointsProvider'
 import { RegionInfo } from './regionInfo'
 import { RegionProvider } from './regionProvider'
 
-export interface RawRegion {
-    description: string
-}
+export class DefaultRegionProvider implements RegionProvider {
+    private readonly onRegionProviderUpdatedEmitter: vscode.EventEmitter<void> = new vscode.EventEmitter()
+    private _loadedRegions: RegionInfo[] = []
 
-export interface RawPartition {
-    partition: string
-    regions: {
-        [regionKey: string]: RawRegion
+    public constructor(endpointsProvider: EndpointsProvider) {
+        endpointsProvider.onEndpointsUpdated(e => this.loadFromEndpointsProvider(e))
+        this.loadFromEndpointsProvider(endpointsProvider)
```

Comment:
Is this necessary with the line above?

Reply:
It is, because there is no guarantee that provider hasn't started or completed loading prior to this constructor being called.